lib/galaxy/config.py

```diff
@@ -1196,3 +1206,14 @@ def _configure_models(self, check_migrate_databases=False, check_migrate_tools=F
     def _configure_signal_handlers(self, handlers):
         for sig, handler in handlers.items():
             signal.signal(sig, handler)
+
+    def _wait_for_database(self, url):
+        from sqlalchemy_utils import database_exists
+        attempts = self.config.database_wait_attempts
+        pause = self.config.database_wait_sleep
+        for i in range(1, attempts):
+            try:
+                database_exists(url)
```

Comment:
Does this need a break here so the loop doesn't continue if the database exists?

Reply:
Of course! (yikes... missed this one!) Thank you! Pushing...
The rest - done, no longer WIP (config schema updated, make config rebuild done).